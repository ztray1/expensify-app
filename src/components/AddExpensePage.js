import React from "React";
import {connect} from "react-redux";
import ExpenseForm from "./ExpenseForm";
import {addExpense} from "../actions/expenses";

export class AddExpensePage extends React.Component{
    submit=(expense)=>{
        this.props.addExpense(expense);
        this.props.history.push("/");
        console.log(this.props);
    };
    render(){
        return(
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm onSubmit={this.submit} />
            </div>
        );
    }
}

const mapDispatchToProps=(dispatch)=>({
    addExpense:(expense)=>dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);