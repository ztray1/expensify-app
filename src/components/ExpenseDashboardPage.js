import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenselistFilters from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";

const ExpenseDashboardPage=()=>(
    <div>
        <ExpensesSummary />
        <ExpenseList />
        <ExpenselistFilters />
    </div>
);
export default ExpenseDashboardPage;