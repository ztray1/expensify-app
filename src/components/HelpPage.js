import React from 'react';

const HelpPage=()=>(
    <div>
        This is form HelpPage
    </div>
);

export default HelpPage;