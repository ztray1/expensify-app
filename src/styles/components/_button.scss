.button{
    background:$blue;
    border:none;
    color:white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight:300;
    line-height: 1;
    padding:$s-size;
    text-decoration: none;
}

.button--link{
    background: none;
}

.button--secondary{
    background: #888;
}

.button--modal{
    margin:0 1rem 0 1rem;
}
