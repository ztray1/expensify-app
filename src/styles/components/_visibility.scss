.show-for-mobile{
    @media(min-width:$desktop-breakpoint){
        dsiplay:none;
    }
}


.show-for-desktop{
    @media(max-width:$desktop-breakpoint - .01rem){
        display: none;
    }
}