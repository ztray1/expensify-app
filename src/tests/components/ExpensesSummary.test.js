import React from "react";
import {shallow} from "enzyme";
import {ExpensesSummary} from "../../components/ExpensesSummary";

test("should correctly render ExpenseSummary with 1 expense",()=>{
     const wrapper=shallow(<ExpensesSummary expenseCount={1} expenseTotal={235} />);
     expect(wrapper).toMatchSnapshot();
});

test("should correctly render ExpenseSummary with mutiple expenses",()=>{
    const wrapper=shallow(<ExpensesSummary expenseCount={123} expenseTotal={23512456} />);
    expect(wrapper).toMatchSnapshot();
});