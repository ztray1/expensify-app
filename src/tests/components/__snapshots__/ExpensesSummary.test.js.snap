// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with mutiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    123
     
    expenses
     totalling 
    $235,124.56
  </h1>
</div>
`;
